enhancements/security/openshift-image-policy.md

```diff
@@ -0,0 +1,511 @@
+
+However, [the signing systems are orthogonal](#overlap-with-openpgp-signing), so technically we can run both in parallel or [change our minds](#changing-our-minds) and decide to refocus on OpenPGP or some new third signing method, as our understanding evolves.
+This proposal covers the early stages of bringing in the Sigstore policy, and later work will cover things like removing OpenPGP to complete [a transitional overlap phase](#overlap-with-openpgp-signing).
+
+### User Stories
+
+* As a cluster administrator, I want to trust that the OpenShift release image I'm running comes from Red Hat, checking whenever a pod is launched from such an image, and not just at update-time.
+* As a cluster administrator, I want to be able to declare my own additional image policies, including restricting the use of the `quay.io/openshift-release-dev` scope.
+* As a Red Hat CI job maintainer or quality engineer, I have a way to disable Sigstore verification while testing releases before they have been signed.
+
+### Goals
+
+* Define an initial OpenShift ClusterImagePolicy that will be installed in all 4.17 clusters to restrict `quay.io/openshift-release-dev/ocp-release` to Red-Hat-signed images.
+
+### Non-Goals
+
+There are a number of possible next-steps that I'm punting on for now, as I try to nibble off the smallest possible slice of progress.
+This enhancement will not:
+
+* Figure out how to configure a ClusterImagePolicy for verifying a mirrored release image, although this will need to happen before we can use Sigstore signatures in disconnected/restricted-network environments.
+* Suggest changes to mirroring tools, although they will need to learn to mirror Sigstore signatures for use in disconnected/restricted-network environments.
+* Suggest changes to the ImageStream controller, although it will need to learn to mirror Sigstore signatures for verification when running pods from ImageStream sources, e.g. [must-gather][].
+* Suggest policies for `quay.io/openshift-release-dev/ocp-v4.0-art-dev`, although those would help tighten down security for other `openshift-*` namespaces.
+* Support testing unsigned releases as HyperShift hosted clusters on a management cluster that requires signed releases.
+
+## Proposal
+
+This enhancement proposes a new `openshift` ClusterImagePolicy that requires Red Hat signatures for `quay.io/openshift-release-dev/ocp-release` images.
+The new ClusterImagePolicy will be managed by the ClusterVersion operator and applied to all clusters by default (initially tech-preview, gated by the `ImagePolicy` feature gate ([currently `TechPreviewNoUpgrade`][feature-gate]).
+
+Internal test clusters, which need to run releases that are not signed by official Red Hat keys, will use a new install-config property `internalTestingImagePolicy: Disable` (also gated by the `ImagePolicy` feature gate, [precedent for a feature-gated install-config property][install-config-feature-gate-precedent]).
```

Comment:
>`internalTestingImagePolicy: Disable`

The name sounds like we're disabling the internal-testing image policy. If we're going with an install config field, what about just `imagePolicy: Disable`? 

But I think the larger question is do we want to expose the knob in the install config? If we're not intending for users to set this for production installs, I think another mechanism like an env var would be preferred.

Reply:
I've pushed e29ce4b3 -> 7aa3d164, pushing the install-config knob down into an alternative section and pivoting to an installer env. var.